public/Scripts/Batalha/batalha_fim.php

```diff
@@ -344,47 +345,83 @@
                 $lvl_mais_forte_vencedor = $pers["lvl"];
             }
         }
-
-        $reputacao = calc_reputacao($vencedor["reputacao"], $perdedor["reputacao"], $lvl_mais_forte_vencedor, $lvl_mais_forte_perdedor);
-        $reputacao_mensal = calc_reputacao($vencedor["reputacao_mensal"], $perdedor["reputacao_mensal"], $lvl_mais_forte_vencedor, $lvl_mais_forte_perdedor);
-
-        if ($usuario["pvp"]["tipo"] == 2) {
-            $reputacao["vencedor_rep"] /= 2;
-            $reputacao["perdedor_rep"] /= 2;
-
-            $reputacao_mensal["vencedor_rep"] /= 2;
-            $reputacao_mensal["perdedor_rep"] /= 2;
-
-            $porcentagem = 0.1;
-            if ($aumento = $userDetails->buffs->get_efeito_from_tripulacao("aumento_berries_saque", $vencedor["id"])) {
-                $porcentagem += $aumento;
+        $verificaLutaAnterior = $connection->run("SELECT COUNT(*) AS total 
+    FROM tb_combate_log 
+    WHERE (id_1 = ? AND id_2 = ?) OR (id_1 = ? AND id_2 = ?)",
+  "iiii", array(
+    $vencedor["id"], 
+    $perdedor["id"], 
+    $perdedor["id"], 
+    $vencedor["id"]
+  ))->fetch_array()["total"];
+
+    
+    if ($verificaLutaAnterior > 0) {
+        // Os participantes já lutaram antes
+        // Obtenha os detalhes da última luta para calcular a reputação
+        $ultimaLuta = $connection->run("SELECT * 
+   FROM tb_combate_log 
+   WHERE (id_1 = ? AND id_2 = ?) OR (id_1 = ? AND id_2 = ?)
+   ORDER BY horario DESC
+   LIMIT 1",
+ "iiii", array(
+    $vencedor["id"], 
+    $perdedor["id"], 
+    $perdedor["id"], 
+    $vencedor["id"]
+ ))->fetch_array();
+
+    
+        // Verifica se o vencedor é o mesmo que ganhou a última luta
+        if ($vencedor['id'] == $vencedor['id']) {
+            // Se o vencedor for o mesmo que o vencedor da última luta, ele não ganha reputação novamente
+            $vencedor_rep = 0;
+            $vencedor_rep_mensal = 0;
+            
+            // Verifica se o perdedor foi o mesmo da última luta
+            if ($vencedor['id'] != $perdedor['id']) {
```

Comment:
esse if é desnecessario, se o vencedor for o meso da ultima luta o perdedor não perde nada de rep é só colocar direto $perdedor_rep = 0 sem esse if else